IDEAS/Buildings/Components/RectangularZoneTemplate.mo

```diff
@@ -557,12 +629,47 @@ protected
     fraType(present=fraTypC.present, U_value=fraTypC.U_value),
     linExtRad=linExtRadWin,
     nWin=nWinC) if
-       hasWinC
+       hasWinC constrainedby Window(inc=IDEAS.Types.Tilt.Wall,
+    glazing(
+      nLay=glazingC.nLay,
+      mats=glazingC.mats,
+      SwAbs=glazingC.SwAbs,
+      SwTrans=glazingC.SwTrans,
+      SwAbsDif=glazingC.SwAbsDif,
+      SwTransDif=glazingC.SwTransDif,
+      U_value=glazingC.U_value,
+      g_value=glazingC.g_value),
+    A=A_winC,
+    frac=fracC,
+    azi=aziA + Modelica.Constants.pi,
+    T_start=T_start,
+    linIntCon_a=linIntCon,
+    dT_nominal_a=dT_nominal_win,
+    linExtCon=linExtCon,
+    windowDynamicsType=windowDynamicsType,
+    redeclare IDEAS.Buildings.Components.Shading.Shading shaType(shaPro(
+  controlled=shaTypC.controlled,
+  shaType=shaTypC.shaType,
+  hWin=shaTypC.hWin,
+  wWin=shaTypC.wWin,
+  wLeft=shaTypC.wLeft,
+  wRight=shaTypC.wRight,
+  ovDep=shaTypC.ovDep,
+  ovGap=shaTypC.ovGap,
+  hFin=shaTypC.hFin,
+  finDep=shaTypC.finDep,
+  finGap=shaTypC.finGap,
+  L=shaTypC.L,
+  dh=shaTypC.dh,
+  shaCorr=shaTypC.shaCorr)),
+    fraType(present=fraTypC.present, U_value=fraTypC.U_value),
+    linExtRad=linExtRadWin)
     "Window for face C of this zone" annotation (Placement(
         transformation(
         extent={{-5,-10},{5,10}},
         rotation=0,
         origin={-95,-30})));
+  replaceable
   IDEAS.Buildings.Components.Window winD(inc=IDEAS.Types.Tilt.Wall, azi=aziA +
```

Comment:
https://github.com/open-ideas/IDEAS/pull/892 has to be merged first

Reply:
This is problematic has we want to by able to redeclare the windows but the way it is coded now has  a constrainedby with all variables filled in. The solution is to make a partial rectangularZoneTemplate which contains everything except the windows. The LinRectangularZoneTemplate with then use directly the LinWindow (no need for redeclaration).